Add Moamen and K-Subarrays to Sortings Bronze (#2122)
cpinitiative/usaco-guide: content/2_Bronze/Intro_Sorting.mdx

</JavaSection>

<PySection>

> Python does not have Static Arrays. See Dynamic Arrays for information on sorting a list.

Reviewer: ```suggestion
Python does not have static arrays. See "Dynamic Arrays" for information on sorting a list.
```

unless there is a reason to have it capitalized? and is dynamic arrays a module?

Reviewer 2: pretty sure he's referring to the section underneath
![image](https://user-images.githubusercontent.com/8647920/130401906-44a802fb-16b4-4f9f-8de9-8b72a6de874b.png)

Reviewer 2: that being said though, i think "static arrays" and "dynamic arrays" should be lowercase

<PySection>

> In order to sort a list, use the `arr.sort()` function. To get a copy of the list in a sorted order, use `sorted(arr)`

Reviewer 2: ```suggestion
In order to sort a list, use the `arr.sort()` function. To get a copy of the list in a sorted order, use `sorted(arr)`.
```

> In order to sort a list, simply use the `arr.sort()` function. To get a copy of the list in a sorted order, use `sorted(arr)`

Reviewer 2: ```suggestion
In order to sort a list, use the `arr.sort()` function. To get a copy of the list in a sorted order, use `sorted(arr)`
```

	
```python
arr = [5, 1, 3, 2, 4]
arr.sort()
print(arr)  # [1, 2, 3, 4, 5]
	
arr = [5, 1, 3, 2, 4]
sorted_arr = sorted(arr) 
print(arr)  # [5, 1, 3, 2, 4]
print(sorted_arr)  # [1, 2, 3, 4, 5]
```

> - [Sorting Basics](https://docs.python.org/3/howto/sorting.html)

Reviewer 2: Not sure if this line is needed since none of the other languages have something like this

Author: It is useful to mention. There are probably times when you would need a copy of a sorted array.

</LanguageSection>

<Warning>

> In order to sort arrays in a descending order, or in any custom order, you'll need to define your own **custom comparator**. This is covered in

Reviewer: i mean you don't necessarily have to, just that it might not be the most efficient

Reviewer 3: what's that supposed to mean

Reviewer 2: you can multiply all the elements by -1 and sort them normally, which will give you the reverse order. not very efficient but it's technically another way to do it

Reviewer 2: Also not sure if a warning block is the best here, maybe an info block is better

Reviewer 4: ^ info block :+1:

> In order to efficiently sort arrays in a descending order, or in any custom order, you'll need to define your own **custom comparator**. This is covered in

Reviewer 4: couldn't sorting and then reversing work for bronze?

Reviewer 4: just for descending order

Author: that would be less efficient than sorting with a custom comp. But yeah, for descending order you could iterate in other ways.

Reviewer 4: the text implies that this makes it impossible I think, so it would be beneficial to reword this

Reviewer 2: ^^ my thoughts exactly

Author: Ok I'll make the below changes?
```suggestion
In order to efficiently sort arrays in a custom order, you'll need to define your own **custom comparator**. For example, instead of reversing a sorted array to get a descending sorted array, you could use a custom comparator. This is covered in
```

Reviewer 4: you don't "need to define your own custom comparator" but it does make the process a lot easier (for most problems), which is what I'm trying to get at. Something along the lines of: "The easiest method of sorting in a custom order is the custom comparator" should suffice.

Reviewer 4: bump @Aadit-Ambadkar

Reviewer 4: .

Author: Done
